src/plugin.js

```diff
@@ -1,69 +1,80 @@
+      allCaps: Joi.boolean().when('camelCase', { is: true, then: Joi.invalid(true) }),
+      firstCap: Joi.boolean().when('camelCase', { is: true, then: Joi.invalid(true) }),
+      deleteOriginal: Joi.boolean().optional()
     }),
     Joi.object().keys({
-      lower_case: Joi.boolean().when('all_caps', { is: true, then: Joi.invalid(true) }),
-      camel_case: Joi.boolean().when('all_caps', { is: true, then: Joi.invalid(true) }),
-      all_caps: Joi.boolean().required(),
-      first_cap: Joi.boolean().when('all_caps', { is: true, then: Joi.invalid(true) }),
-      delete_original: Joi.boolean().optional()
+      lowerCase: Joi.boolean().when('allCaps', { is: true, then: Joi.invalid(true) }),
+      camelCase: Joi.boolean().when('allCaps', { is: true, then: Joi.invalid(true) }),
+      allCaps: Joi.boolean().required(),
+      firstCap: Joi.boolean().when('allCaps', { is: true, then: Joi.invalid(true) }),
+      deleteOriginal: Joi.boolean().optional()
     }),
     Joi.object().keys({
-      lower_case: Joi.boolean().when('first_cap', { is: true, then: Joi.invalid(true) }),
-      camel_case: Joi.boolean().when('first_cap', { is: true, then: Joi.invalid(true) }),
-      all_caps: Joi.boolean().when('first_cap', { is: true, then: Joi.invalid(true) }),
-      first_cap: Joi.boolean().required(),
-      delete_original: Joi.boolean().optional()
+      lowerCase: Joi.boolean().when('firstCap', { is: true, then: Joi.invalid(true) }),
+      camelCase: Joi.boolean().when('firstCap', { is: true, then: Joi.invalid(true) }),
+      allCaps: Joi.boolean().when('firstCap', { is: true, then: Joi.invalid(true) }),
+      firstCap: Joi.boolean().required(),
+      deleteOriginal: Joi.boolean().optional()
     })
   );
 
   var validate = Joi.validate(options, schemaOptions);
   if (validate.error) {
-    return next(validate.error);
+    return Promise.reject(validate.error);
   }
 
   plugin.ext('onRequest', function(request, reply) {
     if (!request.query || Object.keys(request.query).length === 0) {
-      return reply.continue();
+      return endPreHandler(reply);
     }
-    Object.keys(request.query).forEach(function(attr){
+    Object.keys(request.query).forEach(function(attr) {
       var tmpAttr;
       var str;
-      if (options.lower_case) {
+      if (options.lowerCase) {
         //convert FirstName => firstname
         tmpAttr = attr;
         str = attr.toString().toLowerCase();
       }
-      else if (options.camel_case) {
-        // convert FirstName => firstName
-        // convert First_Name => firstName
-        // convert first_name => firstName
+      else if (options.camelCase) {
+        // convert to ot-schema camelCase equivelant or retain original casing.
         tmpAttr = attr;
-        str = attr.toString().replace(/(\_[a-z])/gi, function($1){return $1.toUpperCase().replace('_','');});
-        str = str.substr(0,1).toLowerCase() + str.slice(1,str.length);
+        str = attr.toString().toLowerCase();
```

Comment:
so wouldnt this not convert First_Name => firstName?

Reply:
and then I saw the test below, mmm, is changing this functionality okay?

Reply:
Yes it woud not convert it because First_Name is no in ot-schemas.

The camelCase option was never really working, but I can create a new option called "otSchemaCase" and retain the old camelCase?

Reply:
Don't create otSchemaCase - the old camelCase option never worked, and no one is using it (probably)